hat/assets/js/apps/Iaso/domains/orgUnits/groups/components/Filters.tsx

```diff
@@ -0,0 +1,199 @@
+import React, {
+    FunctionComponent,
+    useState,
+    useMemo,
+    useCallback,
+} from 'react';
+import { Grid, Button } from '@mui/material';
+import SearchIcon from '@mui/icons-material/Search';
+import { useSafeIntl } from 'bluesquare-components';
+import { useGetProjectsDropdownOptions } from '../../../projects/hooks/requests';
+import { useDataSourceVersions } from '../../../dataSources/requests';
+import { useGetDataSources } from '../../../dataSources/useGetDataSources';
+import { useFilterState } from '../../../../hooks/useFilterState';
+import InputComponent from '../../../../components/forms/InputComponent';
+import { baseUrl } from '../config';
+import MESSAGES from '../messages';
+
+type Params = {
+    page?: string;
+    search?: string;
+    dataSource?: string;
+    version?: string;
+    project_ids?: string;
+    accountId: string;
+};
+
+type Props = {
+    params: Params;
+};
+
+const dataSourceByProjectId = (dataSources: any[], projectId: string) => {
```

Comment:
The name of the method should be changed for example: filterDataSourcesByProjectId. And I think it can be simplified like this proposition: 
`const dataSourceByProjectId = (dataSources: any[], projectId: string) => {
    const allDataSource = dataSources?.map(({ projects, name, id }) => {
        return {
            label: name,
            value: `${id}`,
            projectIds: projects?.flat().map(project => project.id.toString()),
        };
    });
    return projectId
        ? allDataSource?.filter(({ projectIds }) =>
              projectIds?.includes(projectId),
          )
        : allDataSource;
};`